mllib/src/test/scala/org/apache/spark/ml/classification/DecisionTreeClassifierSuite.scala

```diff
@@ -275,6 +274,40 @@ class DecisionTreeClassifierSuite extends SparkFunSuite with MLlibTestSparkConte
     val model = dt.fit(df)
   }
 
+  test("DecisionTree should support all NumericType labels") {
+    val dfWithIntLabels = TreeTests.setMetadata(sqlContext.createDataFrame(Seq(
```

Comment:
It might be less verbose to create the dataframe once, and then add the other label column types to the same data frame. Something like:

```scala
val dfWithTypes = df
      .withColumn("shortLabel", df("labelIndex").cast(ShortType))
      .withColumn("longLabel", df("labelIndex").cast(LongType))
      .withColumn("intLabel", df("labelIndex").cast(IntegerType))
      .withColumn("floatLabel", df("labelIndex").cast(FloatType))
      .withColumn("decimalLabel", df("labelIndex").cast(DecimalType(10, 0)))
```

Then just change the label column between training. I'm not sure which way is better, but this would reduce copying the code ~5 times per test.